kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/Type.java

```diff
@@ -24,4 +42,36 @@
 
     String toString( Object value );
 
+    public static Type determineTypeFromClass( Class<?> clazz ) {
+        if( clazz == null ) {
+            return BuiltInType.UNKNOWN;
+        } else if( Number.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.NUMBER;
+        } else if( String.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.STRING;
+        } else if( LocalDate.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.DATE;
+        } else if( LocalTime.class.isAssignableFrom(clazz) || OffsetTime.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.TIME;
+        } else if( ZonedDateTime.class.isAssignableFrom(clazz) || OffsetDateTime.class.isAssignableFrom(clazz) || LocalDateTime.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.DATE_TIME;
+        } else if( Duration.class.isAssignableFrom(clazz) || Period.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.DURATION;
+        } else if( Boolean.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.BOOLEAN;
+        } else if( UnaryTest.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.UNARY_TEST;
+        } else if( Range.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.RANGE;
+        } else if( FEELFunction.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.FUNCTION;
+        } else if( List.class.isAssignableFrom(clazz) ) {
+            return BuiltInType.LIST;
+        } else if( Map.class.isAssignableFrom(clazz) ) {     // TODO not so sure about this one..
```

Comment:
I think we will have to adopt a convention for for this, as we discussed before. We could define that all MapBackedTypes have a key/value pair: "__TYPE__" -> "<TypeName>" . If a given map instance does not have that, it is treated as a context, otherwise we resolve the type by using this key/value pair. What do you think?